examples/scripts/TreeCompareAuto.C

```diff
@@ -0,0 +1,309 @@
+        Bool_t leafIsIdentical = CheckEntriesIdentical(hist, outstream);
+
+        // If the entries are not identical check if the histograms are
+        // identical. This is the case if the entries in the tree are sorted
+        // differently
+        if (leafIsIdentical) {
+            numIdenticalEntries++;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        } else {
+            outstream << "CHecking for identical histograms" << endl;
+            leafIsIdentical = CheckHistogramIdentical(htemp, hist1, outstream);
+        }
+
+        if (leafIsIdentical) {
+            numIdenticalHistograms++;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        } else {
+            outstream << "CHecking Kolmogorov" << endl;
+            leafIsIdentical = CheckHistogramKolmogorov(htemp, hist1, outstream);
+        }
+
+        if (leafIsIdentical) {
+            numKolmogorovHistograms++;
+            outstream << "**********************" << std::endl;
+            hist1->Clear();
+            hist->Clear();
+            htemp->Clear();
+            continue;
+        } else {
+            outstream << "Data are differnt" << std::endl;
+            outstream << "**********************" << std::endl;
+            outstream << "Entries: " << hist->GetEntries() << std::endl;
+            outstream << "Mean: " << hist->GetMean() << std::endl;
+            outstream << "RMS: " << hist->GetRMS() << std::endl;
+            outstream << "Underflow: " << hist->GetBinContent(0) << std::endl;
+            outstream << "Overflow: " << hist->GetBinContent(hist->GetNbinsX() + 1) << std::endl;
+            outstream << "**********************" << std::endl;
+            okay = kFALSE;
+            numFailedLeaves++;
+        }
+    }
+
+    if (!okay) {
+        std::cout << outstream.str();
+        std::cout << "Test failed." << std::endl;
+        std::cout << numFailedLeaves << " of " << numTestedLeaves << " leaves are different." << std::endl;
+        return 1;
+    }
+    //  std::cout << outstream.str();
+    std::cout << "Tested leaves:                    " << numTestedLeaves << std::endl;
+    //  std::cout << "Empty leaves:                     " << numEmptyLeaves << std::endl;
+    std::cout << "Leaves with identical entries:    " << numIdenticalEntries << std::endl;
+    std::cout << "Leaves with identical histograms: " << numIdenticalHistograms << std::endl;
+    std::cout << "Leaves with kolmo histograms:     " << numKolmogorovHistograms << std::endl;
+    std::cout << "Test passed. All leaves of all branches are exactly identical." << std::endl;
+
+    return 0;
+}
+
+Bool_t CheckEntriesIdentical(TH1* compHist, std::stringstream& outstream)
```

Comment:
```suggestion
Bool_t CheckEntriesIdentical(TH1& compHist, std::stringstream& outstream)
```

You're assuming that `compHist != nullptr` anyway. So let's use a reference?

Reply:
Done.